library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/impl/StaticResourceManager.java

```diff
@@ -0,0 +1,176 @@
+	private final ResourceType type;
+	private final LinkedHashMap<String, NamespaceResourceManager> namespaceManagers = new LinkedHashMap<>();
+	private final ArrayList<ResourcePack> resourcePacks = new ArrayList<>();
+	public StaticResourceManager(ResourceType type, List<ResourcePack> packs){
+		this.type = type;
+		this.resourcePacks.addAll(List.copyOf(packs));
+		computeNamespaces();
+	}
+
+	@Override
+	public Set<String> getAllNamespaces() {
+		return ImmutableSet.copyOf(namespaceManagers.keySet());
+	}
+
+	public List<ResourcePack> getPacks(){
+		return ImmutableList.copyOf(resourcePacks);
+	}
+
+	@Override
+	public List<Resource> getAllResources(Identifier id) {
+		ResourceManager resourceManager = this.namespaceManagers.get(id.getNamespace());
+		return resourceManager != null ? resourceManager.getAllResources(id) : List.of();
+	}
+
+	@Override
+	public Map<Identifier, Resource> findResources(String startingPath, Predicate<Identifier> pathFilter) {
+		checkTrailingDirectoryPath(startingPath);
+		Map<Identifier, Resource> map = new TreeMap<>();
+
+		for(NamespaceResourceManager namespaceResourceManager : this.namespaceManagers.values()) {
+			map.putAll(namespaceResourceManager.findResources(startingPath, pathFilter));
+		}
+
+		return map;
+	}
+	@Override
+	public Map<Identifier, List<Resource>> findAllResources(String startingPath, Predicate<Identifier> pathFilter) {
+		checkTrailingDirectoryPath(startingPath);
+		Map<Identifier, List<Resource>> map = new TreeMap<>();
+
+		for(NamespaceResourceManager namespaceResourceManager : this.namespaceManagers.values()) {
+			map.putAll(namespaceResourceManager.findAllResources(startingPath, pathFilter));
+		}
+		return map;
+	}
+
+	@Override
+	public Stream<ResourcePack> streamResourcePacks() {
+		return this.resourcePacks.stream();
+	}
+
+	@Override
+	public Optional<Resource> getResource(Identifier identifier) {
+		ResourceManager resourceManager = this.namespaceManagers.get(identifier.getNamespace());
+		return resourceManager != null ? resourceManager.getResource(identifier) : Optional.empty();
+	}
+
+	@Override
+	public Resource getResourceOrThrow(Identifier identifier) throws FileNotFoundException {
+		Optional<Resource> getResourceResult = getResource(identifier);
+		if(getResourceResult.isEmpty()) throw new FileNotFoundException("No resource found at id " + identifier);
+		else return getResourceResult.get();
+	}
+
+	@Override
+	public InputStream open(Identifier identifier) throws IOException {
+		return AutoCloseableResourceManager.super.open(identifier);
+	}
+
+	@Override
+	public BufferedReader openAsReader(Identifier identifier) throws IOException {
+		return AutoCloseableResourceManager.super.openAsReader(identifier);
+	}
+
+	@Override
+	public void close() {
+		this.resourcePacks.forEach(ResourcePack::close);
+	}
+
+	/**
+	 * Based heavily on {@link MultiPackResourceManagerMixin#quilt$recomputeNamespaces()}
+	 */
+	private void computeNamespaces(){
```

Comment:
again, seems like a lot of moving parts for what's essentially a copy of a little-used feature from vanilla.